host/tests/common/chain.rs

```diff
@@ -0,0 +1,112 @@
+use raiko_lib::consts::{Network, SupportedChainSpecs};
+use rand::Rng;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct RPCResult<T> {
+    pub(crate) result: T,
+}
+
+pub(crate) type BlockHeightResponse = RPCResult<String>;
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct Block {
+    #[serde(rename = "gasUsed")]
+    pub(crate) gas_used: String,
+}
+
+pub(crate) type BlockResponse = RPCResult<Block>;
+
+// NOTE: In order to avoid request collision during multiple tests running in parallel,
+//       we select a random block number to make the proof request unique.
+pub async fn randomly_select_block(network: Network) -> anyhow::Result<u64> {
+    let supported_chains = SupportedChainSpecs::default();
+    let client = reqwest::Client::new();
+    let beacon = supported_chains
+        .get_chain_spec(&network.to_string())
+        .unwrap()
+        .rpc;
+
+    let tip_block_number = get_block_number(network).await?;
+    let random_block_number = rand::thread_rng().gen_range(1..tip_block_number);
+
+    for block_number in random_block_number..tip_block_number {
+        if is_zero_gas_used_block(&client, &beacon, block_number).await? {
```

Comment:
why 0 gas used block?

Reply:
Nice shot. It intends to filter out the empty-logs blocks, which causes error "No BlockProposed event found for block". But 0 gas used is not a good way to do that. I will update it soon.

Reply:
you need to parse the anchor tx to get the corresponding L1 block number to avoid such error, you can find similar logic in previous test.

Reply:
I couldn't find any test case parse anchor tx right now. well, I implement a workaround to do that, PTAL: https://github.com/taikoxyz/raiko/pull/431/commits/b9de365cc9ec121262fe35f206ec9dbbbdeedb9c#diff-57fcfd9f1a07f54b35d3f2bd66f1f2ef036048ed14ac7f6b554472c4edc25eb0R43-R47